x-pack/test/apm_api_integration/tests/alerts/error_count_threshold.spec.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ApmRuleType } from '@kbn/apm-plugin/common/rules/apm_rule_types';
+import { errorCountMessage } from '@kbn/apm-plugin/common/rules/default_action_message';
+import { apm, timerange } from '@kbn/apm-synthtrace-client';
+import expect from '@kbn/expect';
+import { ApmDocumentType } from '@kbn/apm-plugin/common/document_type';
+import { RollupInterval } from '@kbn/apm-plugin/common/rollup';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { createAlertingRule, createIndexConnector } from './alerting_api_helper';
+import { waitForRuleStatus, waitForDocumentInIndex } from './wait_for_rule_status';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+
+  const supertest = getService('supertest');
+  const log = getService('log');
+  const es = getService('es');
+  const apmApiClient = getService('apmApiClient');
+
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  registry.when('error count threshold alert', { config: 'basic', archives: [] }, () => {
+    const start = '2021-01-01T00:00:00.000Z';
+    const end = '2021-01-01T00:15:00.000Z';
+
+    let ruleId: string | undefined;
+    let actionId: string | undefined;
+
+    const INDEX_NAME = 'error-count';
+
+    async function fetchAlertCountOnApm() {
+      const dateRange = {
+        start: new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString(),
+        end: new Date(Date.now() + 5 * 60 * 1000).toISOString(),
+      };
+      const [serviceInventoryResponse, alertsCount] = await Promise.all([
+        apmApiClient.readUser({
+          endpoint: 'GET /internal/apm/services',
+          params: {
+            query: {
+              ...dateRange,
+              environment: 'ENVIRONMENT_ALL',
+              kuery: '',
+              probability: 1,
+              documentType: ApmDocumentType.ServiceTransactionMetric,
+              rollupInterval: RollupInterval.SixtyMinutes,
+            },
+          },
+        }),
+        apmApiClient.readUser({
+          endpoint: 'GET /internal/apm/services/{serviceName}/alerts_count',
+          params: {
+            path: {
+              serviceName: 'service-a',
+            },
+            query: {
+              ...dateRange,
+              environment: 'ENVIRONMENT_ALL',
+            },
+          },
+        }),
+      ]);
+
+      return {
+        serviceInventory: serviceInventoryResponse.body.items[0].alertsCount,
+        alertsCount: alertsCount.body.alertsCount,
+      };
+    }
```

Comment:
This will be useful to other rules as well. Fine to keep here for now, but if you open a new PR with API tests for another rule you should look into extracting it to alerting_api_helper.ts

Reply:
Btw. It would be great if you retrieved the alert counts for all services in the inventory and not just the first one:

```ts
{
  "opbeans-node": 5,
  "opbeans-python": 0
}
```

This way you can add a test that validates that only services with active alerts have a non-zero alert count.